lib/jekyll/filters/url_filters.rb

```diff
@@ -23,14 +22,23 @@ def absolute_url(input)
       # Returns a URL relative to the domain root as a String.
       def relative_url(input)
         return if input.nil?
-        site = @context.registers[:site]
-        return ensure_leading_slash(input.to_s) if site.config["baseurl"].nil?
+        return ensure_leading_slash(input.to_s) if sanitized_baseurl.nil?
         Addressable::URI.parse(
-          ensure_leading_slash(site.config["baseurl"]) + ensure_leading_slash(input.to_s)
+          ensure_leading_slash(sanitized_baseurl) + ensure_leading_slash(input.to_s)
         ).normalize.to_s
       end
 
       private
+
+      def site
+        @context.registers[:site]
+      end
+
+      def sanitized_baseurl
+        baseurl = site.config["baseurl"]
+        return baseurl unless baseurl.nil? || (baseurl == "/")
```

Comment:
And I guess I haven't really checked what currently happens, but what if similarly there is some silliness going on in the site URL part with slashes. Can we robustly handle this in both potential locations?